packages/twenty-front/src/modules/object-record/record-merge/components/MergeRecordsContainer.tsx

```diff
@@ -0,0 +1,207 @@
+import styled from '@emotion/styled';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+
+import { ShowPageContainer } from '@/ui/layout/page/components/ShowPageContainer';
+import { RightDrawerProvider } from '@/ui/layout/right-drawer/contexts/RightDrawerContext';
+import { TabList } from '@/ui/layout/tab-list/components/TabList';
+import { activeTabIdComponentState } from '@/ui/layout/tab-list/states/activeTabIdComponentState';
+import { TabListComponentInstanceContext } from '@/ui/layout/tab-list/states/contexts/TabListComponentInstanceContext';
+import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+
+import { useMergeManyRecords } from '@/object-record/hooks/useMergeManyRecords';
+import { mergeSettingsState } from '@/object-record/record-merge/states/mergeSettingsState';
+import { MergeRecordsTabId } from '@/object-record/record-merge/types/MergeRecordsTabId';
+import { useUpsertRecordsInStore } from '@/object-record/record-store/hooks/useUpsertRecordsInStore';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
+import { useEffect, useState } from 'react';
+import { useNavigateApp } from '~/hooks/useNavigateApp';
+import { useMergeRecordsContainerTabs } from '../hooks/useMergeRecordsContainerTabs';
+import { MergePreviewTab } from './MergePreviewTab';
+import { MergeRecordsFooter } from './MergeRecordsFooter';
+import { MergeRecordTab } from './MergeRecordTab';
+import { MergeSettingsTab } from './MergeSettingsTab';
+
+import { AppPath } from '@/types/AppPath';
+import { useLingui } from '@lingui/react/macro';
+
+const StyledShowPageRightContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: start;
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+`;
+
+const StyledTabList = styled(TabList)`
+  background-color: ${({ theme }) => theme.background.secondary};
+  padding-left: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledContentContainer = styled.div<{ isInRightDrawer: boolean }>`
+  flex: 1;
+  overflow-y: auto;
+  background: ${({ theme }) => theme.background.primary};
+  padding-bottom: ${({ theme, isInRightDrawer }) =>
+    isInRightDrawer ? theme.spacing(16) : 0};
+`;
+
+type MergeRecordsContainerProps = {
+  componentInstanceId: string;
+  objectNameSingular: string;
+  selectedRecords: ObjectRecord[];
+  loading?: boolean;
+  isInRightDrawer?: boolean;
+};
+
+export const MergeRecordsContainer = ({
+  componentInstanceId,
+  objectNameSingular,
+  selectedRecords,
+  loading = false,
+  isInRightDrawer = false,
```

Comment:
A lot of prop drilling for isInRightDrawer, In think we can get rid of the parameter here and reduce the number of occurences of isInRightDrawer as it will always be true